### PR TITLE
generate regions with voronoi

### PR DESCRIPTION
**Pull request by Shinigami92**

### CODE REVIEW

File: Assets/ChunkGenerator.cs [hunk 1]
```diff
@@ -118,8 +123,69 @@ public Chunk GenerateChunk(Vector3Int position)
...
+			return region;
+		}
+
+		byte[] buf = BitConverter.GetBytes(position.x).Concat(BitConverter.GetBytes(position.y)).ToArray();
+		uint valueX = XXHash.CalculateHash(buf, buf.Length, seed: Convert.ToUInt32(seed));
+		uint valueZ = XXHash.CalculateHash(buf, buf.Length, seed: Convert.ToUInt32(seed + 1));
+		Vector2 relativeVoronoiPoint = new Vector2(valueX, valueZ) / uint.MaxValue;
+		Vector2 voronoiPoint = (position + relativeVoronoiPoint);
+		float humidity = (float)SimplexNoise.Scale(NoiseGenerator.Eval(voronoiPoint.x, voronoiPoint.y), 0, 1);
+		float temperature = (float)SimplexNoise.Scale(NoiseGenerator.Eval(voronoiPoint.y, voronoiPoint.x), 0, 1);
```
Comment (Shinigami92) [inline Assets/ChunkGenerator.cs:139]: Note that I changed y and x so that I do not get the same values as for the humidity

File: Assets/Chunk.cs [hunk 2]
```diff
@@ -53,13 +55,24 @@ private void Start()
 					low: 0,
 					high: 16
 				);
+				/*
+				 * TODO: We can optimize this later by first checking all 4 chunk pages.
+				 * If all are in the same region, all blocks in that chunk are in the same region.
+				 */
+				float deltaVoronoi = (float)ChunkGenerator.NoiseGenerator
+					.Eval(noiseX / 4, noiseZ / 4) * 32;
```
Comment (Shinigami92) [inline Assets/Chunk.cs:63]: The divider 4 for scaling and the multiplier 32 for the value are magic numbers that produce a good result

File: Assets/XXHash.cs [hunk 3]
```diff
@@ -0,0 +1,333 @@
+/*
```
Comment (Shinigami92) [inline Assets/XXHash.cs:1]: @ST-DDT Can you please check if we are allowed to copy this file into our project this way?

Comment (ST-DDT) [reply Assets/XXHash.cs:1]: AFAICT its OK to do it like this.

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**